### ISSUE TITLE
Question: Why the GAN loss was not used? Typo?

### ISSUE THREAD

**Issue by hejm37** [senguptaumd/Background-Matting#51]
Hi,
In the train_real_fixed.py, loss_ganG is not used at all? I found in `compose_image_withshift()` in functions.py, `image_sh` is wrapped by `torch.autograd.Variable()`, which will detach it from previous autograph, preventing gradient  propagation from loss_ganG to the generator.

Thank you so much! Should I change that to 
```
def compose_image_withshift(alpha_pred,fg_pred,bg,seg):

    image_sh=torch.zeros(fg_pred.shape).cuda()

    for t in range(0,fg_pred.shape[0]):
        al_tmp=to_image(seg[t,...]).squeeze(2)
        where = np.array(np.where((al_tmp>0.1).astype(np.float32)))
        x1, y1 = np.amin(where, axis=1)
        x2, y2 = np.amax(where, axis=1)

        #select shift
        n=np.random.randint(-(y1-10),al_tmp.shape[1]-y2-10)
        #n positive indicates shift to right
        alpha_pred_sh=torch.cat((alpha_pred[t,:,:,-n:],alpha_pred[t,:,:,:-n]),dim=2)
        fg_pred_sh=torch.cat((fg_pred[t,:,:,-n:],fg_pred[t,:,:,:-n]),dim=2)

        alpha_pred_sh=(alpha_pred_sh+1)/2

        image_sh[t,...]=fg_pred_sh*alpha_pred_sh + (1-alpha_pred_sh)*bg[t,...]

    # return torch.autograd.Variable(image_sh.cuda())
    return image_sh
```

**Comment by mozpp** [comment]
will torch.autograd.Variable() cut the back-propagation?

**Comment by mozpp** [comment]
> will torch.autograd.Variable() cut the back-propagation?

it will
